lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -0,0 +1,75 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    # Rack module includes middlewares that are required to trace any framework
+    # and application built on top of Rack.
+    module Rack
+      # TraceMiddleware ensures that the Rack Request is properly traced
+      # from the beginning to the end. The middleware adds the request span
+      # in the Rack environment so that it can be retrieved by the underlying
+      # application. If request tags are not set by the app, they will be set using
+      # information available at the Rack level.
+      class TraceMiddleware
+        DEFAULT_CONFIG = {
+          tracer: Datadog.tracer,
+          default_service: 'rack'
+        }.freeze
+
+        def initialize(app, options = {})
+          # access tracer configurations
+          user_settings = DEFAULT_CONFIG.merge(options)
+          @app = app
+          @tracer = user_settings.fetch(:tracer)
+          @service = user_settings.fetch(:default_service)
+
+          # configure the Rack service
+          @tracer.set_service_info(
+            @service,
+            'rack',
+            Datadog::Ext::AppTypes::WEB
```

Comment:
Side note, wondering if, at some point we could have something like a `Proxy` type. It's a generic pattern, after all, just, handling & routing things for the servers doing the real job. But just a side note, this is very fine for now.